// @angular/material/menu/menu-trigger.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
import * as import0 from '@angular/material/menu/menu-trigger';
import * as import1 from '@angular/core/src/change_detection/change_detection';
import * as import3 from '@angular/core/src/linker/view_utils';
export var Wrapper_MdMenuTrigger = (function () {
    function Wrapper_MdMenuTrigger(p0, p1, p2, p3, p4) {
        this._changed = false;
        this.context = new import0.MdMenuTrigger(p0, p1, p2, p3, p4);
        this._expr_0 = import1.UNINITIALIZED;
    }
    Wrapper_MdMenuTrigger.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_MdMenuTrigger.prototype.ngOnDestroy = function () {
        this.context.ngOnDestroy();
        (this.subscription0 && this.subscription0.unsubscribe());
        (this.subscription1 && this.subscription1.unsubscribe());
    };
    Wrapper_MdMenuTrigger.prototype.check_menu = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this._changed = true;
            this.context.menu = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_MdMenuTrigger.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        return changed;
    };
    Wrapper_MdMenuTrigger.prototype.checkHost = function (view, componentView, el, throwOnChange) {
    };
    Wrapper_MdMenuTrigger.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        if ((eventName == 'mousedown')) {
            var pd_sub_0 = (this.context._handleMousedown($event) !== false);
            result = (pd_sub_0 && result);
        }
        if ((eventName == 'click')) {
            var pd_sub_1 = (this.context.toggleMenu() !== false);
            result = (pd_sub_1 && result);
        }
        return result;
    };
    Wrapper_MdMenuTrigger.prototype.subscribe = function (view, _eventHandler, emit0, emit1) {
        this._eventHandler = _eventHandler;
        if (emit0) {
            (this.subscription0 = this.context.onMenuOpen.subscribe(_eventHandler.bind(view, 'onMenuOpen')));
        }
        if (emit1) {
            (this.subscription1 = this.context.onMenuClose.subscribe(_eventHandler.bind(view, 'onMenuClose')));
        }
    };
    return Wrapper_MdMenuTrigger;
}());
