// @angular/material/tabs/ink-bar.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
import * as import0 from '@angular/material/tabs/ink-bar';
export var Wrapper_MdInkBar = (function () {
    function Wrapper_MdInkBar(p0, p1) {
        this._changed = false;
        this.context = new import0.MdInkBar(p0, p1);
    }
    Wrapper_MdInkBar.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_MdInkBar.prototype.ngOnDestroy = function () {
    };
    Wrapper_MdInkBar.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        return changed;
    };
    Wrapper_MdInkBar.prototype.checkHost = function (view, componentView, el, throwOnChange) {
    };
    Wrapper_MdInkBar.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        return result;
    };
    Wrapper_MdInkBar.prototype.subscribe = function (view, _eventHandler) {
        this._eventHandler = _eventHandler;
    };
    return Wrapper_MdInkBar;
}());
