// @angular/material/tabs/tab-label-wrapper.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
import * as import0 from '@angular/material/tabs/tab-label-wrapper';
import * as import1 from '@angular/core/src/change_detection/change_detection';
import * as import3 from '@angular/core/src/linker/view_utils';
export var Wrapper_MdTabLabelWrapper = (function () {
    function Wrapper_MdTabLabelWrapper(p0, p1) {
        this._changed = false;
        this.context = new import0.MdTabLabelWrapper(p0, p1);
        this._expr_0 = import1.UNINITIALIZED;
        this._expr_1 = import1.UNINITIALIZED;
    }
    Wrapper_MdTabLabelWrapper.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_MdTabLabelWrapper.prototype.ngOnDestroy = function () {
    };
    Wrapper_MdTabLabelWrapper.prototype.check_disabled = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this._changed = true;
            this.context.disabled = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_MdTabLabelWrapper.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        return changed;
    };
    Wrapper_MdTabLabelWrapper.prototype.checkHost = function (view, componentView, el, throwOnChange) {
        var currVal_1 = this.context.disabled;
        if (import3.checkBinding(throwOnChange, this._expr_1, currVal_1)) {
            view.renderer.setElementClass(el, 'md-tab-disabled', currVal_1);
            this._expr_1 = currVal_1;
        }
    };
    Wrapper_MdTabLabelWrapper.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        return result;
    };
    Wrapper_MdTabLabelWrapper.prototype.subscribe = function (view, _eventHandler) {
        this._eventHandler = _eventHandler;
    };
    return Wrapper_MdTabLabelWrapper;
}());
